src/Umbraco.Web/PropertyEditors/ContentPickerPropertyEditor.cs

```diff
@@ -33,17 +35,97 @@ protected override PreValueEditor CreatePreValueEditor()
 
         internal class ContentPickerPreValueEditor : PreValueEditor
         {
+            private const string _startNodeAlias = "startNodeId";
+
             [PreValueField("showOpenButton", "Show open button", "boolean")]
             public string ShowOpenButton { get; set; }
 
             [PreValueField("showEditButton", "Show edit button (this feature is in preview!)", "boolean")]
             public string ShowEditButton { get; set; }
-            
-            [PreValueField("startNodeId", "Start node", "treepicker")]
+
+            [PreValueField(_startNodeAlias, "Start node", "treepicker")]
             public int StartNodeId { get; set; }
 
             [PreValueField("showPathOnHover", "Show path when hovering items", "boolean")]
             public bool ShowPathOnHover { get; set; }
+
+            public override IDictionary<string, PreValue> ConvertEditorToDb(IDictionary<string, object> editorValue, PreValueCollection currentValue)
+            {
+                var processedEditorValues = new Dictionary<string, PreValue>();
+                foreach (var preval in editorValue)
+                {
+                    var processedPreVal = new KeyValuePair<string, PreValue>(preval.Key, preval.Value == null ? null : new PreValue(preval.Value.ToString()));
+                    if (preval.Key == _startNodeAlias && preval.Value != null)
+                    {
+                        int startNodeUniqueId;
+                        if (int.TryParse(preval.Value.ToString(), out startNodeUniqueId))
+                        {
+                            // look up the node
+                            var cs = ApplicationContext.Current.Services.ContentService;

```

Comment:
You are essentially just looking up Ids + GUIDs, you aren't really working with content in which case you should use the EntityService which would be much faster to lookup these types of things, in fact there are methods in there to just do this int->guid conversion and vice versa. 

You should not be using `ApplicationContext.Current` singleton at all in this code. You should be injecting the service that you require into this pre-value class's constructor (i.e. the `IEntityService`) so that this code can be unit tested.  The class responsible for creating this class is the `ContentPickerPropertyEditor`. This class should be given an empty ctor (the default) and a new ctor that has the parameters required to fulfill the dependencies for creating this pre-value editor. Then in the empty ctor you'd pass in the services required based on the singleton. For example:

```
private readonly IEntityService _entityService;

public ContentPickerPropertyEditor(IEntityService entityService)
{
    if (entityService == null) throw new ArgumentNullException("entityService");
    _entityService = entityService;
    _internalPreValues = new Dictionary<string, object>
    {
        {"startNodeId", "-1"},
        {"showOpenButton", "0"},
        {"showEditButton", "0"},
        {"showPathOnHover", "0"}
    };
}

public ContentPickerPropertyEditor(): this(ApplicationContext.Current.Services.EntityService)
{
}
```

Then you can pass in `_entityService` to the ctor of this pre value editor class. 

In v8, Property Editors (among every other plugin type) have Dependency Injection built in, so you would always have a ctor that is expecting all dependencies and they will automatically be wired up.